tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -21,599 +21,601 @@
+
+    char *test_vec[] = {
+        simple_uri_00,
+        simple_uri_01,
+        simple_uri_02,
+        simple_uri_03,
+        simple_uri_04,
+        simple_uri_05,
+        simple_uri_07,
+        simple_uri_08,
+        simple_uri_09,
+        simple_uri_10,
+        simple_uri_11,
+        simple_uri_12,
+        simple_uri_17,
+        simple_uri_19,
+        simple_uri_20,
+        simple_uri_21,
+        simple_uri_22,
+        simple_uri_23,
+        simple_uri_24,
+        simple_uri_25,
+        simple_uri_26,
+        simple_uri_27,
+        simple_uri_28,
+        simple_uri_29,
+        simple_uri_30,
+        simple_uri_31,
+        simple_uri_32,
+        simple_uri_33,
+        simple_uri_34,
+        simple_uri_36,
+        simple_uri_37,
+        simple_uri_38,
+        simple_uri_39,
+    };
+
+    uri_parser_result_t uri_res;
+    int ret = 0;
+
+    for (unsigned int i = 0; i < ARRAY_SIZE(test_vec); ++i) {
+        ret = uri_parser_process_string(&uri_res, test_vec[i]);
+        if (ret != expected_ret) {
+            _print_return_expectation(test_vec[i], expected_ret, ret);
+            TEST_FAIL(failure_msg);
+        }
+    }
+}
+
+/*
+ * Given a simple invalid uri, this test checks that the parsing returns an error
+ */
+static void _successful_rejecting_of_invalid_uri(void)
+{
+    char *failure_msg = "Failure: The uri_parser failed to reject an invalid uri.";
+    int expected_ret = -1;  /* Indicates an invalid uri */
+
+    /* these are the old test vectors containing invalid uri */
+    char *simple_uri_06 = "coap://RIOT:test@[fe80:db8::1%]:5683/.well-known/core?v=1";
+    char *simple_uri_13 = "coap://R@[2001:db8::1]:5own/v=1";
+    char *simple_uri_14 = "coap://R@[2001:db8::1]:5own/:v=1";
+    char *simple_uri_15 = "cap://R@[2001:db8::1]:5own/?v=1";
+    char *simple_uri_16 = "oap://Y2001:db8::1]:5own/av=1";
+    char *simple_uri_18 = "coap://oap://P@[2001:b";
+    char *simple_uri_35 = "";
+    char *simple_uri_40 = "coap://example.com:1234568910";
+    char *simple_uri_41 = "coap://example.com: 12/foo";
+    char *simple_uri_42 = "coap://example.com:0x12/foo";
+
+    char *test_vec[] = {
+        simple_uri_06,
+        simple_uri_13,
+        simple_uri_14,
+        simple_uri_15,
+        simple_uri_16,
+        simple_uri_18,
+        simple_uri_35,
+        simple_uri_40,
+        simple_uri_41,
+        simple_uri_42,
+    };
+
+    uri_parser_result_t uri_res;
+    int ret = 0;
+
+    for (unsigned int i = 0; i < ARRAY_SIZE(test_vec); ++i) {
+        ret = uri_parser_process_string(&uri_res, test_vec[i]);
+        if (ret != expected_ret) {
+            _print_return_expectation(test_vec[i], expected_ret, ret);
+            TEST_FAIL(failure_msg);
         }
     }
 }
 
+/*
+ * This test checks if the uri_parser returns an error if the port length
+ * (as a string) is invalid, that is, longer than 5 characters.
+ * (technically, the port in an uri can be zero to infit characters long; ABNF: *DIGIT)
```

Comment:
Not sure I get what the "technical" side note is explaining. Nevertheless, it would be "infinit**e** number of"